iwf/errors.py

```
@@ -112,9 +112,8 @@ def process_workflow_abnormal_exit_error(
 
 
 def parse_unexpected_error(err) -> ErrorResponse:
-    encoding = guess_json_utf(err.content)
-    if encoding is not None:
-        err_dict = jsonlib.loads(err.content.decode(encoding))
-    else:
-        err_dict = jsonlib.loads(err.content)
-    return ErrorResponse.from_dict(err_dict)
+    try:
+        response = Response(200, content=err.content)
+        return ErrorResponse.from_dict(response.json())
+    except jsonlib.decoder.JSONDecodeError:
```

Should it be just `except` to match the if-else logic?

---

Makes sense, fixed now! ✅

---

Getting an error in the pipeline due to this change. Is there a way that we can suppress this error just for this specific line?

```
warning: The top-level linter settings are deprecated in favour of their counterparts in the `lint` section. Please update the following options in `pyproject.toml`:
  - 'ignore' -> 'lint.ignore'
iwf/errors.py:118:5: E722 Do not use bare `except`
    |
116 |         response = Response(err.status_code, content=err.content)
117 |         return ErrorResponse.from_dict(response.json())
118 |     except:
    |     ^^^^^^ E722
119 |         return ErrorResponse.from_dict(jsonlib.loads(err.content))
```

---

Try `except Exception`

https://stackoverflow.com/a/54948581

---

If it makes sense. My concern is what is the fallback case. Try fails, except catches a different Exception than `jsonlib.decoder.JSONDecodeError`, what happens then?

---

That's a fair point. I'm trying to see if I can reproduce a case where an Exception would be thrown here, but I can't seem to find one. 

I even tried to go to the `main` branch to see if the old fallback case was being covered, and it wasn't. Also, the unit tests don't seem to include such scenario. 

Due to this, what I'm going to do is remove the fallback

---

I think we should keep it as close to the original code as possible since I'm not sure of the side effects.

```
try

except Exception
```

Feels like the best match to me. Does linter like it?

---

Linter is ok with this approach. The IDE displays a warning about it still being too broad, but looks like it's ok to do.

---

Also - changed this to have `except Exception` now ✅